purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoHelper.kt

```diff
@@ -0,0 +1,183 @@
+package com.revenuecat.purchases
+
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.strings.ConfigureStrings
+import com.revenuecat.purchases.strings.CustomerInfoStrings
+
+internal class CustomerInfoHelper(
+    private val deviceCache: DeviceCache,
+    private val backend: Backend,
+    private val identityManager: IdentityManager,
+    private val handler: Handler = Handler(Looper.getMainLooper())
+) {
+
+    var updatedCustomerInfoListener: UpdatedCustomerInfoListener? = null
```

Comment:
So I was debating with myself on whether to use the same pattern we use with the `PurchasesState` here and create something like `CustomerInfoHelperState` or something like that to hold the listener and the `lastSentCustomerInfo`. Maybe making the state immutable would be better, but I believe it should also work like this, since access to both properties should be synchronized. I went with the simple approach for now. Lmk if you think we should mimic that pattern cc @vegaro

Reply:
Separated from that, I was wondering if we've thought about using `WeakReference` to hold the listener. That way we avoid possible leaks in our users. That said, this might break some of our users implementations, so might not be a good idea.